scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala

```diff
@@ -84,14 +85,31 @@ package object jdbc {
     def apply[T](jdbcIoOptions: JdbcIoOptions): JdbcIO[T] = JdbcIO[T](jdbcIoId(jdbcIoOptions))
   }
 
-  private def jdbcIoId(opts: JdbcConnectionOptions, query: String): String =
-    s"${opts.username}:${opts.password}@${opts.connectionUrl}:$query"
+  private def jdbcIoId(opts: JdbcConnectionOptions, query: String): String = {
+    val user = if (opts.password.isDefined) {
+      s"${opts.username}:${opts.password}"
+    } else {
+      s"${opts.username}"
+    }
+    s"$user@${opts.connectionUrl}:$query"
+  }
+
 
   private def jdbcIoId(opts: JdbcIoOptions): String = opts match {
     case JdbcReadOptions(connOpts, query, _, _) => jdbcIoId(connOpts, query)
     case JdbcWriteOptions(connOpts, statement, _) => jdbcIoId(connOpts, statement)
   }
 
+  private def getDataSourceConfig(connOpts: jdbc.JdbcConnectionOptions): DataSourceConfiguration = {
+    val dataSourceConfig = jio.JdbcIO.DataSourceConfiguration
+      .create(connOpts.driverClass.getCanonicalName, connOpts.connectionUrl)
+      .withUsername(connOpts.username)
+    if (connOpts.password.isDefined) {
+      dataSourceConfig.withPassword(connOpts.password.get)
```

Comment:
not sure if there is a better way to do this than calling `.get`

Reply:
don't see any downside as`.get` is safeguard with `if` check

Reply:
Maybe `connOpts.password.foreach(p => dataSourceConfig.withPassword(p))`